cmds/cbmem/main.go

```diff
@@ -0,0 +1,350 @@
+// Copyright 2016-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// dirname prints out the directory name of one or more args.
+// If no arg is given it returns an error and prints a message which,
+// per the man page, is incorrect, but per the standard, is correct.
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"reflect"
+
+	_ "github.com/google/netstack/tcpip/header"
+)
+
+/*
+
+import (
+    "golang.org/x/text/language"
+    "golang.org/x/text/message"
+)
+
+func main() {
+    p := message.NewPrinter(language.English)
+    p.Printf("%d\n", 1000)
+
+    // Output:
+    // 1,000
+}
+*/
+
+var (
+	cbmem struct {
+		Memory           *memoryEntry
+		MemConsole       *memconsoleEntry
+		Consoles         []string
+		TimeStamps       []timestampEntry
+		UART             []serialEntry
+		MainBoard        mainboardEntry
+		Hwrpb            hwrpbEntry
+		CBMemory         []cbmemEntry
+		BoardID          BoardIDEntry
+		StringVars       map[string]string
+		BootMediaParams  bootMediaParamsEntry
+		VersionTimeStamp uint32
+		Unknown          []uint32
+		Ignored          []string
+	}
+
+	TimeStampNames = map[int]string{
```

Comment:
this map should be in the types.go file, I think, or a const.go or something.

Reply:
const.go for now.